Modules/Core/Common/include/itkImageRegionRange.h

```diff
@@ -146,52 +146,50 @@ class ImageRegionRange final
     {}
 
     template <size_t VIndex>
-    void Increment(std::true_type) noexcept
+    void
+    Increment() noexcept
     {
-      static_assert(VIndex < (ImageDimension - 1), "For a larger index value, the other overload should be picked");
-
       m_BufferIterator += m_OffsetTable[VIndex];
 
-      if (static_cast<SizeValueType>(++m_IterationOffset[VIndex]) >= m_IterationRegionSize[VIndex])
+      if constexpr (VIndex < (ImageDimension - 1))
```

Comment:
Thanks for reminding me we can now use this!

Some other things which we can now use:
https://www.codingame.com/playgrounds/2205/7-features-of-c17-that-will-simplify-your-code/introduction

Reply:
That article suggests that instances of `enable_if` are candidates for replacement with `if constexpr`.

Reply:
Thanks for your encouragement. In the meantime I found two cases of "old-style compile-time _if_" by std::true_type/std::false_type overloading: in [itkImageHelper.h](https://github.com/InsightSoftwareConsortium/ITK/blob/b7bfef30a8729118131db908e1c423aeb734c219/Modules/Core/Common/include/itkImageHelper.h). Preparing another commit for this PR to address those as well, please wait  😃  If you see other possible use cases for _constexpr if_, please just try it out!